src/olympia/api/serializers.py

```diff
@@ -0,0 +1,93 @@
+    fake_object) is populated with the ES data in order to work well with
+    the parent model serializer (e.g., AddonSerializer).
+
+    """
+    # In base classes add the field names we want converted to Python
+    # date/datetime from the Elasticsearch date strings.
+    datetime_fields = ()
+
+    def __init__(self, *args, **kwargs):
+        super(BaseESSerializer, self).__init__(*args, **kwargs)
+
+        # Set all fields as read_only just in case.
+        for field_name in self.fields:
+            self.fields[field_name].read_only = True
+
+        if getattr(self, 'context'):
+            for field_name in self.fields:
+                self.fields[field_name].context = self.context
+
+    def get_fields(self):
+        """
+        Return all fields as normal, with one exception: replace every instance
+        of TranslationSerializerField with ESTranslationSerializerField.
+        """
+        fields = super(BaseESSerializer, self).get_fields()
+        for key, field in fields.items():
+            if isinstance(field, TranslationSerializerField):
+                fields[key] = ESTranslationSerializerField(source=field.source)
+        return fields
+
+    def to_native(self, data):
+        obj = self.fake_object(data)
+        return super(BaseESSerializer, self).to_native(obj)
+
+    def fake_object(self, data):
+        """
+        Create a fake model instance from ES data which serializer fields will
+        source from.
+        """
+        raise NotImplementedError
+
+    def _attach_fields(self, obj, data, field_names):
+        """Attach fields to fake instance."""
+        for field_name in field_names:
+            value = getattr(data, field_name, None)
+            if field_name in self.datetime_fields:
+                value = self.to_datetime(value)
+            setattr(obj, field_name, value)
+        return obj
+
+    def _attach_translations(self, obj, data, field_names):
+        """Deserialize ES translation fields."""
+        for field_name in field_names:
+            ESTranslationSerializerField.attach_translations(
+                obj, data, field_name)
+        return obj
+
+    def to_datetime(self, value):
+        """
+        Returns a datetime given an Elasticsearch date/datetime field.
+        """
+        if not value or isinstance(value, (date, datetime)):
+            return
+
+        if len(value) == 26:
+            try:
+                return datetime.strptime(value, '%Y-%m-%dT%H:%M:%S.%f')
+            except (TypeError, ValueError):
+                pass
+        elif len(value) == 19:
+            try:
+                return datetime.strptime(value, '%Y-%m-%dT%H:%M:%S')
+            except (TypeError, ValueError):
+                pass
+        elif len(value) == 10:
+            try:
+                return datetime.strptime(value, '%Y-%m-%d')
+            except (TypeError, ValueError):
+                pass
+
+        return value
```

Comment:
I forgot to add tests, yeah.

This was stolen from Marketplace, not sure what kind of data we have as I'm not using it yet. I'll add tests and figure out a better way to deal with the fields.